internal/addrs/module.go

```diff
@@ -168,3 +171,107 @@ func (m Module) Ancestors() []Module {
 func (m Module) configMoveableSigil() {
 	// ModuleInstance is moveable
 }
+func (m Module) configRemovableSigil() {}
+
+func parseModulePrefix(traversal hcl.Traversal) (Module, hcl.Traversal, tfdiags.Diagnostics) {
+	remain := traversal
+	var module Module
+	var diags tfdiags.Diagnostics
+
+	for len(remain) > 0 {
+		moduleName, isValidModule, moduleNameDiags := getModuleName(remain)
+		diags = diags.Append(moduleNameDiags)
+
+		if !isValidModule || diags.HasErrors() {
+			break
```

Comment:
instead of breaking, we could probably just return here, and remove the `if len(remain) > 0` later on.

Reply:
I tested this option, and it throws an error. I need to keep [this](https://github.com/opentofu/opentofu/blob/ca1d39b88bf7d9c7522bf5a0c73807aa87d332a1/internal/addrs/module.go#L209-L222) logic. 
Let's say I have the following address `module.example.null_resource.example_resource`.  The `TraverseRoot` in this case is `module`. 
After I parse the module part of the address, I want to return the remaining `null_resource.example_resource`. If I won't set `null_resource` as the new root, later in `ParseRemoveEndpoint()` -> `parseResourceUnderModule()` I won't be able to access the Root [here](https://github.com/opentofu/opentofu/blob/ca1d39b88bf7d9c7522bf5a0c73807aa87d332a1/internal/addrs/parse_target.go#L128) and it'll throw an error instead.